python/mxnet/gluon/model_zoo/text/base.py

```diff
@@ -54,7 +54,7 @@ def forward(self, inputs, begin_state=None):
 def _apply_weight_drop_to_rnn_cell(block, rate, weight_drop_mode = 'training'):
     params = block.collect_params('h2h_weight')
     weight_dropped_params = WeightDropParameter(params['h2h_weight'], rate, mode)
-    block.collect_params('h2h_weight').params['h2h_weight'] = weight_dropped_params
+    block.collect_params('h2h_weight')._params['h2h_weight'] = weight_dropped_params
```

Comment:
what if the cell is created with a parameter dictionary as input to the constructor (i.e. `params=external_params_dict`? In that case the member that actual has the parameter would be in `_shared`.